client/asset/btc/btc.go

```diff
@@ -2361,6 +2370,228 @@ func (btc *baseWallet) fund(keep uint64, minConfs uint32, lockUnspents bool,
+			} else {
+				fundingUTXOs = leastOverFund(enough, remainingUTXOs)
+			}
+			if len(fundingUTXOs) == 0 {
+				return allFundingUTXOs
+			}
+			totalFunded += sumUTXOs(fundingUTXOs)
+			if totalFunded > avail-keep {
+				return allFundingUTXOs
+			}
+			allFundingUTXOs = append(allFundingUTXOs, fundingUTXOs)
+			remainingUTXOs = utxoSetDiff(remainingUTXOs, fundingUTXOs)
+		}
+		return allFundingUTXOs
+	}
+
+	reorderFundingUTXOs := func(fundingUTXOs [][]*compositeUTXO, orderedValues []*asset.MultiOrderValue) [][]*compositeUTXO {
+		reorderedFundingUTXOs := make([][]*compositeUTXO, 0, len(fundingUTXOs))
+		for _, value := range values {
+			for i, orderedValue := range orderedValues {
+				if value == orderedValue {
+					reorderedFundingUTXOs = append(reorderedFundingUTXOs, fundingUTXOs[i])
+					break
+				}
+			}
+		}
+		return reorderedFundingUTXOs
+	}
+
+	returnValues := func(allFundingUTXOs [][]*compositeUTXO) (coins []asset.Coins, redeemScripts [][]dex.Bytes, fundingCoins map[outPoint]*utxo, spents []*output, err error) {
+		coins = make([]asset.Coins, len(allFundingUTXOs))
+		fundingCoins = make(map[outPoint]*utxo)
+		spents = make([]*output, 0, len(allFundingUTXOs))
+		redeemScripts = make([][]dex.Bytes, len(allFundingUTXOs))
+		for i, fundingUTXOs := range allFundingUTXOs {
+			coins[i] = make(asset.Coins, len(fundingUTXOs))
+			redeemScripts[i] = make([]dex.Bytes, len(fundingUTXOs))
+			for j, output := range fundingUTXOs {
+				coins[i][j] = newOutput(output.txHash, output.vout, output.amount)
+				fundingCoins[outPoint{txHash: *output.txHash, vout: output.vout}] = &utxo{
+					txHash:  output.txHash,
+					vout:    output.vout,
+					amount:  output.amount,
+					address: output.address,
+				}
+				spents = append(spents, newOutput(output.txHash, output.vout, output.amount))
+				redeemScripts[i][j] = output.redeemScript
+			}
+		}
+		return
+	}
+
+	valuesCopy := make([]*asset.MultiOrderValue, len(values))
+	copy(valuesCopy, values)
+
+	// Try to fund in increasing order of value first.
+	sort.Slice(valuesCopy, func(i, j int) bool {
+		return valuesCopy[i].Value < valuesCopy[j].Value
+	})
+	allFundingUTXOs := fundInOrder(valuesCopy)
+	if len(allFundingUTXOs) == len(values) { // successfully funded all orders
+		reorderedFundingUTXOs := reorderFundingUTXOs(allFundingUTXOs, valuesCopy)
+		return returnValues(reorderedFundingUTXOs)
+	}
+
+	// If could not fully fund, fund as much as possible in the priority
+	// order.
+	allFundingUTXOs = fundInOrder(values)
+	return returnValues(allFundingUTXOs)
+}
+
+// fundMultiWithSplit creates a split transaction with one output for each
+// order in order to fund the orders. An error is returned if there is not
+// enough funds in the wallet to cover the split transaction for each order.
+func (btc *baseWallet) fundMultiWithSplit(keep, maxLock uint64, values []*asset.MultiOrderValue, splitTxFeeRate, maxFeeRate, splitBuffer uint64) ([]asset.Coins, [][]dex.Bytes, uint64, error) {
```

Comment:
Will you doc `splitBuffer` somewhere, either here or on the tagged options struct.
I don't understand it's purpose or how to set it.